src/tools/rust-analyzer/crates/ide-assists/src/tests/sourcegen.rs

```diff
@@ -56,7 +56,13 @@ r#####"
             assists.into_iter().map(|it| it.to_string()).collect::<Vec<_>>().join("\n\n"),
         );
         let dst = project_root().join("docs/user/generated_assists.adoc");
-        fs::write(dst, contents).unwrap();
+        if let Err(e) = fs::write(&dst, contents) {
+            if std::env::var("CI").is_ok() {
+                // ignore failures
```

Comment:
this seems ... Not great? Why are we ignoring errors? Would like to hear from an RA maintainer that this is ok.

Reply:
These tests are _really not_ suited to Rust CI.

They:

  * Generate source code
  * Run it through `rustfmt +stable` (not available in CI)
  * Ensure that the version on disk matches the version we just generated

It's great when developing locally, and it even makes sense on their own CI, but we can't run them here. I started by stubbing `reformat` (since we don't have rustfmt stable, and using a stageN rustfmt might give different results), and the tests starting failing because the unformatted version doesn't match the formatted version.

I guess disabling all the tests might be the better option here.. I'll ping RA folks about this.

Reply:
Ok. I don't mind that they're disabled here. My main concern is you've disabled them in rust-analyzer's CI too.

Would it be helpful if rust-analyzer set a different environment variables, like RUST_ANALYZER_CI=1 or something?

Reply:
I ended up reverting those two commits and simply disabling those tests when the `in-rust-tree` feature is enabled.

(The downside is that a _bunch_ of crates have an `in-rust-tree` feature now.. and there's a lot of `cfg` attributes now. Jeeze that's a lot of `cfg` attributes.)